<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>
<body>
    <script>
        //网络请求：aaa 自己处理10行
        //网络请求：aaa111 自己处理10行
        //网络请求：aaa111222 自己处理10行
        // new Promise((resolve,reject) => {
        //     setTimeout(() => {
        //         resolve('aaa')
        //     },1000)
        // }).then(res => {
        //     console.log(res,'第一次10行代码');
        //     return new Promise((resolve) => {
        //         resolve(res + '111')
        //     })
        // }).then(res => {
        //     console.log(res,'第二次10行代码');
        //     return new Promise((resolve) => {
        //         resolve(res + '222')
        //     })
        // }).then(res => {
        //     console.log(res,'第3次10行代码');
        // })


        new Promise((resolve,reject) => {
            setTimeout(() => {
                resolve('aaa')
            },1000)
        }).then(res => {
            console.log(res,'第一次10行代码');
            return Promise.resolve(res + 111)
        }).then(res => {
            console.log(res,'第二次10行代码');
            return Promise.resolve(res + 222)
        }).then(res => {
            console.log(res,'第3次10行代码');
        })


        



        // new Promise((resolve,reject) => {
        //     setTimeout(() => {
        //         resolve('aaa')
        //     },1000)
        // }).then(res => {
        //     console.log(res,'第一次10行代码');
        //     return res + 111
        // }).then(res => {
        //     console.log(res,'第二次10行代码');
        //     return res + 222
        // }).then(res => {
        //     console.log(res,'第3次10行代码');
        // })


        // new Promise((resolve,reject) => {
        //     setTimeout(() => {
        //         resolve('aaa')
        //     },1000)
        // }).then(res => {
        //     console.log(res,'第一次10行代码');
        //     // return res + 111
        //     // return Promise.reject('err message')
        //     throw  'err message'
        // }).then(res => {
        //     console.log(res,'第二次10行代码');
        //     return res + 222
        // }).then(res => {
        //     console.log(res,'第3次10行代码');
        // }).catch(err => {
        //     console.log(err);
        // })
    </script>
</body>
</html>